presto-main/src/main/java/com/facebook/presto/operator/SpillableGroupedTopNBuilder.java

```diff
@@ -128,26 +131,19 @@ public WorkProcessor<Page> buildResult()
         checkSpillSucceeded(spillInProgress);
 
         // Convert revocable memory to user memory as returned Iterator holds on to memory so we no longer can revoke.
-        if (!inputInMemoryGroupedTopNBuilder.isEmpty()) {
-            long currentRevocableBytes = localRevocableMemoryContext.getBytes();
-            localRevocableMemoryContext.setBytes(0);
-            if (!localUserMemoryContext.trySetBytes(localUserMemoryContext.getBytes() + currentRevocableBytes)) {
-                localRevocableMemoryContext.setBytes(currentRevocableBytes);
-                checkSpillSucceeded(spillToDisk());
-            }
-        }
-
         if (!spiller.isPresent()) {
-            return inputInMemoryGroupedTopNBuilder.buildResult();
+            if (inputInMemoryGroupedTopNBuilder.isEmpty() || inputInMemoryGroupedTopNBuilder.migrateMemoryContext(localUserMemoryContext)) {
```

Comment:
Want to make sure I understand what's going on here - it looks like previously we would build the result and switch to user memory as long as inputInMemoryGroupedTopNBuilder fit in the user memory, and now we only do it if we haven't started spilling.  Is that correct?

Reply:
Context to understand this change: The current design is that, if we have previously spilled input , then we will spill the last accumulated input regardless of wether it fits into userMem or not.

This code change is just trying to cleanly implement accounting for that design. i.e if we have previously spilled, then do not try to unnecessarily move to localMemory , because we are going to spill it anyway